### ISSUE TITLE
OpenRPC definitions are inaccurate

### ISSUE THREAD

**Issue by aatifsyed** [filecoin-project/lotus#12164]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus fvm/fevm - Lotus FVM and FEVM interactions
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt/WinningPoSt)
- [X] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
1.27.1
```

### Describe the Bug

https://github.com/filecoin-project/lotus/blob/36d96342c36186ab15417b53511dce94b66ae089/build/openrpc/full.json#L71-L91

The schema for, e.g `cid` is wrong - it's not
```json
{
  "type": "string"
}
```
it's
```json
{
  "type": "object",
  "required": ["/"],
  "properties": {
    "/": { "type": "string" }
  }
}
```
(as expected and emitted by lotus)

### Logging Information

```text
N/A
```

**Comment by aatifsyed** [comment]
There may be other bugs in the definitions - forest has put effort into emitting correct schemas which may be a useful reference

**Comment by rvagg** [comment]
Great! Thanks for this one.

What about the use of `TipSetKey`, they seem to be `"type": ["object"]` but could be defined as an array of CIDs, does Forest do this differently?

**Comment by aatifsyed** [comment]
Yep, [forest represents them correctly](https://github.com/ChainSafe/filecoin-common-node-api/blob/97ea8acdcb5edf0a1bd159f2861dd77ea923c337/schemas/forest-45e7c88c97c.json#L188-L195), also respecting the quirk that empty arrays are serialized as `null` by Lotus

```json
{
  "type": [
    "array",
    "null"
  ],
  "items": {
    "$ref": "#/components/schemas/Cid"
  },
  "components": {
    "schemas": {
      "Cid": {
        "type": "object",
        "required": [
          "/"
        ],
        "properties": {
          "/": {
            "type": "string"
          }
        }
      }
    }
  }
}
```

**Comment by aatifsyed** [comment]
- related: #12214
